### ISSUE TITLE
Potential erroneous masking of Werror

### ISSUE THREAD

**Issue by KireinaHoro** [aciceri/rock5b-nixos#4]
I'm going through the kernel patches in `linux-rock5b` and it seems like this is a dangerous one:

https://github.com/aciceri/rock5b-nixos/blob/e1099dd4a6be8018a5cc489fb2ee1d50e4e285a0/packages/kernel/0001-Ignore-implementation-defects-warned-by-newer-GCC.patch#L49

It masks the warning/error on a [undefined reference to a macro `late_initcall`](https://github.com/radxa/kernel/blob/a9583fb47e9b96c552f458a376207141602c0025/drivers/soc/rockchip/iomux.c#LL87C7692-L87C7709), which should be solved by including `<linux/init.h>` instead of simply masking that error:

I'm putting the following patch in my own flake, but it may be useful for you to merge into `linux-rock5b`:

```diff
diff --git a/drivers/soc/rockchip/iomux.c b/drivers/soc/rockchip/iomux.c
index b6af2bd3b..7586f9d6f 100644
--- a/drivers/soc/rockchip/iomux.c
+++ b/drivers/soc/rockchip/iomux.c
@@ -9,6 +9,7 @@
 #include <linux/fs.h>
 #include <linux/list.h>
 #include <linux/uaccess.h>
+#include <linux/init.h>
 #include <linux/ioctl.h>
 #include <linux/types.h>
 #include <linux/miscdevice.h>
```

**Comment by aciceri** [comment]
Thanks for pointing this out! Would you like to open a PR?

Also, since you also use this board, would like joining efforts and moving this repository to `nix-community`?

**Comment by KireinaHoro** [comment]
The pull request is at https://github.com/aciceri/rock5b-nixos/pull/5.

What do I need to do in helping this repo move to `nix-community`?

**Comment by aciceri** [comment]
Thanks again.
For some reason I haven't enabled CI for PRs so I'll merge this blindly trusting that it will build :)
If you are going to play with this device and want to keep your personal config separated by this "agnostic" layer I can give you write permissions on this repo, so things will be faster.

However I want to move this repo to `nix-community` for at least two reasons:
- some people have their own configurarations for this device, I had to suffer many times before finding out that someone else already solve my problems. So I it would be cool if we could all join efforts and centralize useful stuff.
- I want to use [nix-community's infrastructure](https://github.com/nix-community/infra) `aarch64-linux` builders. Now I'm cross-compiling with Github CI but I noticed that when you build the system from the Rock5B itself it rebuilds the kernel. I also tried native compilation with QEMU's binfmt but it was really slow (more than 4 hours to build the kernel).
